pennylane/pauli/pauli_arithmetic.py

```diff
@@ -55,17 +55,30 @@
     Z: matZ,
 }
 
-sparse_matI = sparse.eye(2, format="csr")
-sparse_matX = sparse.csr_matrix([[0, 1], [1, 0]])
-sparse_matY = sparse.csr_matrix([[0, -1j], [1j, 0]])
-sparse_matZ = sparse.csr_matrix([[1, 0], [0, -1]])
 
-sparse_mat_map = {
-    I: sparse_matI,
-    X: sparse_matX,
-    Y: sparse_matY,
-    Z: sparse_matZ,
-}
+@lru_cache
+def _cached_sparse_data(mat):
+    """Returns the sparse data and indices of a Pauli matrix."""
+    if mat == "I":
+        data = np.array([1.0, 1.0], dtype=np.complex128)
+        indices = np.array([0, 1], dtype=np.int64)
```

Comment:
This is likely not hit because none of the PauliWord in the tests is `PauliX(0) ... Identity(N-1)`. We should add such a test.

Reply:
True! I still have to do a thorough pass through the tests, for now I had mainly focused in the implementation. Thanks for pointing this out 😄